editor/block-settings-menu/unknown-converter.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { IconButton } from '@wordpress/components';
+import { getUnknownTypeHandlerName, pasteHandler } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import { getBlock } from '../selectors';
+import { replaceBlocks } from '../actions';
+
+export function UnknownConverter( { block, convertToBlocks } ) {
+	if ( ! block || getUnknownTypeHandlerName() !== block.name ) {
+		return null;
+	}
+
+	return (
+		<IconButton
+			className="editor-block-settings-menu__control"
+			onClick={ () => convertToBlocks( block ) }
```

Comment:
Not entirely sure what the best way is here to avoid passing arguments.

Reply:
What's the desire to avoid passing the argument? If you mean so that the component doesn't need to be aware of a `block` prop, the only way would probably be overriding react-redux `connect`'s third argument [`mergeProps`](https://github.com/reactjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options) for the dispatch to be aware of the block from state, but... probably more work than is really worthwhile.

Reply:
Can you make the component not know about the block? Don't you have to return `null` if the block isn't defined (or has an invalid name)?